src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useRef } from 'react';
+import { StyledCheckbox } from './Checkbox.styles';
+import { CheckboxProps } from './types';
+
+export const Checkbox = ({ checked, disabled, indeterminate, onChange, ...rest }: CheckboxProps) => {
+  const checkboxRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (checkboxRef && checkboxRef.current && indeterminate) {
```

Comment:
This condition will not set `checkBoxRef.current.indeterminate` to false when `indeterminate prop` changes from true to false. Is this wanted why design?

Reply:
Correct😅 fixing it

Reply:
Changed, and added a test for that case. What do you think?